### PR TITLE
test/tools: update staticcheck version to latest

### PR DESCRIPTION
**Pull request by easwars**
RELEASE NOTES: none

### CODE REVIEW

File: test/tools/go.mod [hunk 1]
```diff
@@ -1,17 +1,17 @@
 module google.golang.org/grpc/test/tools
 
-go 1.21
+go 1.22.1
```
Comment (printchard) [inline test/tools/go.mod:3]: Is there a reason we are going to use `1.22.1` instead of `1.22.6`? Just curious about it.

Comment (easwars) [reply test/tools/go.mod:3]: This was not manually edited. I simply ran `go get honnef.co/go/tools` from the `test/tools` directory.

Comment (easwars) [reply test/tools/go.mod:3]: But my guess is that the Go version here got updated because some dependency that was updated here has an updated Go version.

Comment (printchard) [reply test/tools/go.mod:3]: Ah, I see. I thought since `1.23` released we weren't supporting `1.21` anymore. Thanks for the answer!

Comment (easwars) [reply test/tools/go.mod:3]: The `go.mod` file touched in this PR from the `test/tools` directory. Out top-level `go.mod` still says `1.21` here: https://github.com/grpc/grpc-go/blob/master/go.mod

Comment (printchard) [reply test/tools/go.mod:3]: Oh I see. I misunderstood, thanks for explaining!

Comment (dfawley) [reply test/tools/go.mod:3]: It also doesn't really matter since this exists purely for dependency management..nothing builds from here.

**Review by dfawley — CHANGES_REQUESTED** [review]
I'm fine with this, but it triggers a new vet failure now.  Can you take a look?

**Comment by easwars** [comment]
> I'm fine with this, but it triggers a new vet failure now. Can you take a look?

Fixed the failures. Thanks.

**Review by dfawley — APPROVED** [review]

**Pull request merged**